Store current url in hash

barryvdh opened the pull request:
Not sure if this is the most 'Vue-ish' way, but this will store the url in the hash. When the page is loaded with something in the hash, it will trigger the check directly. Eg `http://kupo.app/#http://example.com`

phanan commented:
While it works, the current solution doesn't seem to validate the hash value. How about a more "proper," server-side supported parameter? I'm thinking of:

* Pushing http://kupo.dev/?url=http://urlto.check into history upon form submission
* On the server side, include a `window.defaultUrl = $url` in the rendered `<script>` (`app.blade.php`). This `$url` should obviously be sanitized.
* In Vue's `created`: `this.url = window.defaultUrl`

barryvdh commented:
Alright, used pushState, much better indeed :)

Going back also works with popstate. Added a check for empty url so you can go back to the first page without errors.
Split up the submit function so that we don't pushState every time (beginning and going back etc)

phanan commented:
Absolutely amazing, thanks!
